helm/file_repo_store.go

```diff
@@ -0,0 +1,489 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/banzaicloud/pipeline/config"
+	phelm "github.com/banzaicloud/pipeline/pkg/helm"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"k8s.io/helm/pkg/downloader"
+	"k8s.io/helm/pkg/getter"
+	helmEnv "k8s.io/helm/pkg/helm/environment"
+	"k8s.io/helm/pkg/helm/helmpath"
+	"k8s.io/helm/pkg/repo"
+)
+
+type FileRepositoryStore struct {
+	Env helmEnv.EnvSettings
+}
+
+func NewFileRepositoryStore(orgName string) (*FileRepositoryStore, error) {
+	var helmPath = config.GetHelmPath(orgName)
+	env := createEnvSettings(fmt.Sprintf("%s/%s", helmPath, phelm.HelmPostFix))
+
+	repoStore := &FileRepositoryStore{Env: env}
+	err := repoStore.init()
+	if err != nil {
+		return nil, err
+	}
+	return repoStore, nil
+}
+
+// ReposGet returns repo
+func (s *FileRepositoryStore) ReposGet() ([]*repo.Entry, error) {
+
+	repoPath := s.Env.Home.RepositoryFile()
+	log.Debugf("Helm repo path: %s", repoPath)
+
+	f, err := repo.LoadRepositoriesFile(repoPath)
+	if err != nil {
+		return nil, err
+	}
+	if len(f.Repositories) == 0 {
+		return make([]*repo.Entry, 0), nil
+	}
+
+	return f.Repositories, nil
+}
+
+// ReposAdd adds repo(s)
+func (s *FileRepositoryStore) ReposAdd(Hrepo *repo.Entry) (bool, error) {
+	repoFile := s.Env.Home.RepositoryFile()
+	var f *repo.RepoFile
+	if _, err := os.Stat(repoFile); err != nil {
+		log.Infof("Creating %s", repoFile)
+		f = repo.NewRepoFile()
+	} else {
+		f, err = repo.LoadRepositoriesFile(repoFile)
+		if err != nil {
+			return false, errors.Wrap(err, "Cannot create a new ChartRepo")
+		}
+		log.Debugf("Profile file %q loaded.", repoFile)
+	}
+
+	for _, n := range f.Repositories {
+		log.Debugf("repo: %s", n.Name)
+		if n.Name == Hrepo.Name {
+			return false, nil
+		}
+	}
+
+	c := repo.Entry{
+		Name:  Hrepo.Name,
+		URL:   Hrepo.URL,
+		Cache: s.Env.Home.CacheIndex(Hrepo.Name),
+	}
+	r, err := repo.NewChartRepository(&c, getter.All(s.Env))
+	if err != nil {
+		return false, errors.Wrap(err, "Cannot create a new ChartRepo")
+	}
+	log.Debugf("New repo added: %s", Hrepo.Name)
+
+	errIdx := r.DownloadIndexFile("")
+	if errIdx != nil {
+		return false, errors.Wrap(errIdx, "Repo index download failed")
+	}
+	f.Add(&c)
+	if errW := f.WriteFile(repoFile, 0644); errW != nil {
+		return false, errors.Wrap(errW, "Cannot write helm repo profile file")
```

Comment:
```suggestion
		return false, errors.Wrap(errW, "cannot write helm repo profile file")
```

Error and log messages are generally lower cased.